runtime/common/src/claims.rs

```diff
@@ -193,29 +238,10 @@ decl_module! {
 			ensure_none(origin)?;
```

Comment:
What stops someone with a preclaim + signing for claiming this way and bypass that all?

Reply:
For example, this test passes:

```
	#[test]
	fn bypass_attest_claiming() {
		new_test_ext().execute_with(|| {
			assert_eq!(Balances::free_balance(42), 0);
			let s = sig::<Test>(&dave(), &42u64.encode(), &[]);
			let r = Claims::claim(Origin::NONE, 42, s.clone());
			assert_ok!(r);
			assert_eq!(Balances::free_balance(&42), 200);
			assert_eq!(Claims::total(), 400);
		});
	}
```

In your tests, dave has a statement to sign:

```		GenesisConfig::<Test>{
			claims: vec![
				(eth(&alice()), 100, None, None),
				(eth(&dave()), 200, None, Some(StatementKind::Default)),
				(eth(&eve()), 300, Some(42), Some(StatementKind::Alternative))
			],
```

Reply:
should be sorted now.